src/sysext/sysext.c

```diff
@@ -262,9 +276,33 @@ static int unmerge_hierarchy(
                 if (r == 0)
                         break;
 
+                r = read_one_line_file(work_dir_info_file, &escaped_work_dir);
+                if (r < 0) {
+                        if (r != -ENOENT)
+                                return log_error_errno(r, "Failed to read '%s': %m", work_dir_info_file);
+                } else {
+                        ssize_t l;
+
+                        l = cunescape_length(escaped_work_dir, r, 0, &work_dir);
+                        if (l < 0)
+                                return log_error_errno(l, "Failed to unescape work directory path: %m");
+                }
+
+                r = umount_verbose(LOG_DEBUG, dot_dir, MNT_DETACH|UMOUNT_NOFOLLOW);
+                if (r < 0) {
+                        if (r != -EINVAL)
```

Comment:
maybe add comment `/* not a mount point? */`

Reply:
Wrote some longer explanation.